02/app.js

```diff
@@ -1,7 +1,47 @@
 
 /* rozwiązanie z pętlą for */
-const x = 4;
+// const x = prompt('Podaj liczbę!');
+// const checkMultiplying = (num) => {
+//   for (let y = 1; y <= 9; y++) {
+//     console.log(`${x} x ${y} = ${x * y}`);
+//   }
+// };
 
+// checkMultiplying(x);
 
 
-/* rozwiązanie z pętlą while  */
\ No newline at end of file
+
+
+/* rozwiązanie z pętlą while  */
+let z = 1
+
+const pow = (base, power) => {
+  let result = '';
+  while (z <= power) {
+    result = result.concat(`${base} * `)
+    z++;
+    if (z === power) {
+      return result = result.concat(`${base} = ${Math.pow(base, power)}`)
+    }
```

Comment:
Raczej myślę, że taki zapis niepotrzebnie komplikuje rozwiązanie.
W pętli masz określone kiedy się kończy więc nie widzę sensu robić to ponownie przez `if` + `return`.
Lepiej po prostu wrzucić return poza `while`.